packages/app-desktop/services/spellChecker/SpellCheckerServiceDriverNative.ts

```diff
@@ -17,43 +16,25 @@ export default class SpellCheckerServiceDriverNative extends SpellCheckerService
 		return this.session().availableSpellCheckerLanguages;
 	}
 
-	// Language can be set to '' to disable spell-checking
-	public setLanguage(v: string) {
+	// Language can be set to [] to disable spell-checking
+	public setLanguage(v: string[]) {
 		// If we pass an empty array, it disables spell checking
 		// https://github.com/electron/electron/issues/25228
-		if (!v) {
+		if (v.length == 0) {
 			this.session().setSpellCheckerLanguages([]);
 			return;
 		}
 
-		// The below function will throw an error if the provided language is
-		// not supported, so we provide fallbacks.
-		// https://github.com/laurent22/joplin/issues/4146
-		const languagesToTry = [
-			v,
-			languageCodeOnly(v),
-		].concat(localesFromLanguageCode(languageCodeOnly(v), this.availableLanguages));
-
-		for (const toTry of languagesToTry) {
-			try {
-				this.session().setSpellCheckerLanguages([toTry]);
-				logger.info(`Set effective language from "${v}" to "${toTry}"`);
-				return;
-			} catch (error) {
-				logger.warn(`Failed to set language to "${toTry}". Will try the next one in this list: ${JSON.stringify(languagesToTry)}`);
-				logger.warn('Error was:', error);
-			}
-		}
```

Comment:
Don't we want to keep that or a variation of it? It will crash the app if the language is not recognised and we can't be sure it will be.

Reply:
I found another way to prevent the crash, as I described in commit message and [here](https://github.com/laurent22/joplin/pull/6617#discussion_r908634719).

Long story short, the issue hapens when the app tries to set language for the first time. For that it uses Electron function `getSpellCheckerLanguages`, witch can return language code that don't supported by Electron language checker, especially on macOS. But `availableSpellCheckerLanguages` will always return language codes that are supported be language checker, since they are hard coded in source code.

So in order to check if language is supported or not, we don't need to try set it and see if it fails. We can just check if the language returned by `getSpellCheckerLanguages` is in `availableSpellCheckerLanguages`, and if not set our own default language (en-US). I make this check in `setupDefaultLanguage` in `SpellCheckerService`.

Reply:
Hmm, but then you move the responsibility of not crashing and checking the languages from this function to the caller. In practice it means the languages would have to be checked every time before calling this function. Could you at least move back the checking logic to this function?

Reply:
> In practice it means the languages would have to be checked every time before calling this function.

You don't have to do this, because you get the list of languages from `availableSpellCheckerLanguages` and all languages there is supported by spell checker and won't cause a error. The only time that unsupported language could come up is when using `getSpellCheckerLanguages`, and we using it only one time, when the user first starts the app and don't have spell check languages configured.

I moved it to caller, because we have to set default language and store it to settings if `getSpellCheckerLanguages` returns unsupported language.

Reply:
@antontkv, the fact that you removed this check introduced a regression which (as predicted) crashes the app: https://discourse.joplinapp.org/t/pre-release-v2-9-is-now-available-updated-23-october/26416/29

Are you able to fix this issue?

Reply:
Fixed in ae178016ab26507f580528e6807d758c2815134e